cognite/client/data_classes/postgres_gateway/tables.py

```diff
@@ -0,0 +1,336 @@
+        )
+
+
+@dataclass
+class Column(CogniteObject):
+    name: str
+    type: ColumnType
+
+    @classmethod
+    def _load(cls, resource: dict[str, Any], cognite_client: CogniteClient | None = None) -> Self:
+        return cls(
+            name=resource["propertyName"],
+            type=resource["type"],
+        )
+
+    def dump(self, camel_case: bool = True) -> dict[str, Any]:
+        return {
+            "propertyName": self.name,
+            "type": self.type,
+        }
+
+    @classmethod
+    def _load_columns(cls, data: dict[str, Any]) -> list[Column]:
+        columns: list[Column] = []
+        for name, column_data in data.items():
+            columns.append(
+                Column(
+                    name=name,
+                    type=column_data["type"],
+                )
+            )
+        return columns
+
+    @classmethod
+    def _dump_columns(cls, columns: Sequence[Column]) -> dict[str, Any]:
+        return {column.name: {"type": column.type} for column in columns}
+
+
+class _TableCore(WriteableCogniteResource["TableWrite"], ABC):
+    _type: str
+
+    def __init__(self, tablename: str):
+        self.tablename = tablename
+
+    def dump(self, camel_case: bool = True) -> dict[str, Any]:
+        return {"tablename": self.tablename, "type": self._type}
+
+
+class TableWrite(_TableCore, ABC):
+    """View and create foreign **tables** for a given **user**.
+
+    This is the write/request format of the table.
+    """
+
+    def as_write(self) -> Self:
+        return self
+
+    @classmethod
+    def _load(cls, resource: dict[str, Any], cognite_client: CogniteClient | None = None) -> Self:
+        type_ = resource.get("type")
+        if type_ is None and hasattr(cls, "_type"):
+            type_ = cls._type
+        elif type_ is None:
+            raise KeyError("type")
+        try:
+            return cast(Self, _TABLE_WRITE_CLASS_BY_TYPE[type_]._load_table(resource))
+        except KeyError:
+            raise TypeError(f"Unknown source type: {type_}")
+
+    @classmethod
+    @abstractmethod
+    def _load_table(cls, data: dict[str, Any]) -> Self:
+        raise NotImplementedError()
+
+
+class RawTableWrite(TableWrite):
+    """Foreign tables.
+
+    This is the read/response format of the raw table.
+
+    Args:
+        tablename (str): Name of the foreign table.
+        options (RawTableOptions): Table options
+        columns (Sequence[Column]): Foreign table columns.
+
+    """
+
+    _type = "raw_rows"
+
+    def __init__(self, tablename: str, options: RawTableOptions, columns: Sequence[Column]) -> None:
+        super().__init__(tablename=tablename)
+        self.options = options
+        self.columns = columns
+
+    @classmethod
+    def _load_table(cls, data: dict[str, Any]) -> Self:
+        return cls(
+            tablename=data["tablename"],
+            options=RawTableOptions._load(data["options"]),
+            columns=Column._load_columns(data["columns"]),
```

Comment:
Would it make sense if this was a `ColumnList`?
```py
  columns=ColumnList._load(data["columns"]),
```
What I'm thinking is we could maybe drop the non-standard load and dump methods (`_dump_columns` / `_load_columns`)